packages/start/config/index.js

```diff
@@ -57,6 +57,19 @@ export function defineConfig(baseConfig = {}) {
     entryExtension = ".jsx";
   }
 
+  const { clientPlugins, serverPlugins, serverFunctionsPlugins } =
+    typeof plugins === 'object' && !Array.isArray(plugins)
```

Comment:
should check if plugins is a function, if it is treat as the same as if it was an array

Reply:
`typeof plugins` will return `"function"` if it is.